tests/v2/validation/rbac/rbac_test.go

```diff
@@ -0,0 +1,240 @@
+package rbac
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	log "github.com/sirupsen/logrus"
+)
+
+
+type rbTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUser       *management.User
+	standardUserClient *rancher.Client
+	session            *session.Session
+	cluster            *management.Cluster
+}
+
+
+func (rb *rbTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *rbTestSuite) SetupSuite() {
+	testSession := session.NewSession(rb.T())
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	//Get cluster name from the config file and append cluster details in rb
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *rbTestSuite) ValidateClustersList(client *rancher.Client, expectedClusterCount int, clusterID string) (err error) {
```

Comment:
Can this be extended to Project Owner, Project member and restricted admin user roles also ? 

Also this function name `ValidateClustersList` should be a more appropriate one, since this covers about 8 tests.

Reply:
`ValidateClustersList` does not cover 8 test cases. The purpose of this function is to validate the cluster list obtained as a cluster member/owners/standard users.

Reply:
Yes. This can be extended to Project Owner, project members and restricted admins as well.

Reply:
Quick question, this function is agnostic to the role of the user, correct?

Reply:
yes, its agnostic to the role of the user